### ISSUE TITLE
Update cilium chart to support IPv6 only config

### ISSUE THREAD

**Issue by rbrtbnfgl** [rancher/rke2#3065]
**Is your feature request related to a problem? Please describe.**

Cilium fails to start in case of IPv6 only config
```
write-kubeconfig-mode: 644
cluster-cidr: "2001:cafe:43:0::/56"
service-cidr: "2001:cafe:43:1::/112"
cni: "cilium"
```
**Describe the solution you'd like**

Update Cilium chart to support the IPv6 only configuration

**Describe alternatives you've considered**

**Additional context**

Cilium with IPv6 does not support tunnel mode.
On a multi node setup pod-to-pod communication works if the pods are on the same node in case of pods on different nodes the cluster network has to support the forwarding of the pod IPv6 traffic.

**Comment by VestigeJ** [comment]
Currently on hold for - https://github.com/rancher/rke2/issues/3067

**Comment by VestigeJ** [comment]
TLDR cilium ipv6 only is looking good now - the initial branch this targeted went EOL before the cilium changes could be added in so the previous comment was dead ended. I'm just circling back to show a cluster running with this configuration now.


```
=========== rke2 config ===========
write-kubeconfig-mode: 644
debug: true
token: YOUR_TOKEN_HERE
cni: cilium
profile: cis-1.23
selinux: true
protect-kernel-defaults: true
node-ip: 2600:iced:ab4:iced:coffee:baby:609e:97e
node-external-ip: 2600:iced:ab4:iced:coffee:baby:609e:97e
cluster-cidr: 2001:cafe:42:0::/56
service-cidr: 2001:cafe:42:1::/112
```


```
kubectl get no,po,svc -A -o wide                                                                                                                                                                                         ip-1-1-1-17: Thu Sep  7 18:26:10 2023

NAME                     STATUS   ROLES                       AGE     VERSION          INTERNAL-IP                              EXTERNAL-IP                             OS-IMAGE             KERNEL-VERSION    CONTAINER-RUNTIME
node/ip-229   Ready    control-plane,etcd,master   83s     v1.28.1+rke2r1   2600:iced:ab4:iced:coffee:baby:609e:97e   <none>                                  Ubuntu 20.04.3 LTS   5.11.0-1022-aws   containerd://1.7.3-k3s1
node/ip-17    Ready    control-plane,etcd,master   6m41s   v1.28.1+rke2r1   2600:iced:ab4:iced:coffee:baby:609e:97e    2600:1f1c:ab4:ee48:c9fe:926d:609e:97e   Ubuntu 20.04.3 LTS   5.11.0-1022-aws   containerd://1.7.3-k3s1
node/ip-243   Ready    control-plane,etcd,master   2m58s   v1.28.1+rke2r1   2600:iced:ab4:iced:coffee:baby:609e:97e   <none>                                  Ubuntu 20.04.3 LTS   5.11.0-1022-aws   containerd://1.7.3-k3s1
node/ip-38     Ready    <none>                      2m17s   v1.28.1+rke2r1   2600:iced:ab4:iced:coffee:baby:609e:97e   <none>                                  Ubuntu 20.04.3 LTS   5.11.0-1022-aws   containerd://1.7.3-k3s1

NAMESPACE     NAME                                                       READY   STATUS              RESTARTS        AGE     IP                                       NODE                NOMINATED NODE   READINESS GATES
kube-system   pod/cilium-42ctx                                           1/1     Running             0               6m13s   2600:iced:ab4:iced:coffee:baby:609e:97e    ip-17    <none>           <none>
kube-system   pod/cilium-f9cjn                                           1/1     Running             0               83s     2600:iced:ab4:iced:coffee:baby:609e:97e   ip-229   <none>           <none>
kube-system   pod/cilium-k6kxg                                           1/1     Running             0               2m57s   2600:iced:ab4:iced:coffee:baby:609e:97e   ip-243   <none>           <none>
kube-system   pod/cilium-operator-68f4699b9d-d89jq                       1/1     Running             0               6m13s   2600:iced:ab4:iced:coffee:baby:609e:97e    ip-17    <none>           <none>
kube-system   pod/cilium-operator-68f4699b9d-np8xj                       1/1     Running             0               6m13s   2600:iced:ab4:iced:coffee:baby:609e:97e   ip-243   <none>           <none>
kube-system   pod/cilium-vd45b                                           1/1     Running             0               119s    2600:iced:ab4:iced:coffee:baby:609e:97e   ip-38     <none>           <none>
kube-system   pod/cloud-controller-manager-ip-229             0/1     Running             0               25s     2600:dddd:ab4:ee48:babb:7ba6:bbbb:b7da   ip-229   <none>           <none>
kube-system   pod/cloud-controller-manager-ip-17              0/1     Running             1 (2m27s ago)   6m27s   2600:iced:ab4:iced:coffee:baby:609e:97e    ip-17    <none>           <none>
kube-system   pod/cloud-controller-manager-ip-243             0/1     Running             0               117s    2600:iced:ab4:iced:coffee:baby:609e:97e   ip-243   <none>           <none>
kube-system   pod/etcd-ip-229                                 0/1     Running             0               67s     2600:iced:ab4:iced:coffee:baby:609e:97e   ip-229   <none>           <none>
kube-system   pod/etcd-ip-17                                  0/1     Running             1 (3m7s ago)    6m28s   2600:iced:ab4:iced:coffee:baby:609e:97e    ip-17    <none>           <none>
kube-system   pod/etcd-ip-243                                 0/1     Running             0               2m28s   2600:iced:ab4:iced:coffee:baby:609e:97e   ip-243   <none>           <none>
kube-system   pod/helm-install-rke2-cilium-txtww                         0/1     Completed           0               6m23s   2600:iced:ab4:iced:coffee:baby:609e:97e    ip-17    <none>           <none>
kube-system   pod/helm-install-rke2-coredns-69w8t                        0/1     Completed           0               6m23s   2600:iced:ab4:iced:coffee:baby:609e:97e    ip-17    <none>           <none>
kube-system   pod/helm-install-rke2-ingress-nginx-dr6bx                  0/1     Completed           0               6m23s   2001:cafe:42::a52f                       ip-17    <none>           <none>
kube-system   pod/helm-install-rke2-metrics-server-h7p9j                 0/1     Completed           0               6m22s   2001:cafe:42::2d66                       ip-17    <none>           <none>
kube-system   pod/helm-install-rke2-snapshot-controller-crd-jrkf6        0/1     Completed           0               6m21s   2001:cafe:42::493f                       ip-17    <none>           <none>
kube-system   pod/helm-install-rke2-snapshot-controller-t4gbp            0/1     Completed           1               6m21s   2001:cafe:42::6a41                       ip-17    <none>           <none>
kube-system   pod/helm-install-rke2-snapshot-validation-webhook-lvxn4    0/1     Completed           0               6m20s   2001:cafe:42::6bf4                       ip-17    <none>           <none>
kube-system   pod/kube-apiserver-ip-229                       1/1     Running             0               78s     2600:iced:ab4:iced:coffee:baby:609e:97e   ip-229   <none>           <none>
kube-system   pod/kube-apiserver-ip-17                        1/1     Running             0               6m34s   2600:iced:ab4:iced:coffee:baby:609e:97e    ip-17    <none>           <none>
kube-system   pod/kube-apiserver-ip-243                       1/1     Running             0               2m15s   2600:iced:ab4:iced:coffee:baby:609e:97e   ip-243   <none>           <none>
kube-system   pod/kube-controller-manager-ip-229              0/1     Running             0               25s     2600:iced:ab4:iced:coffee:baby:609e:97e   ip-229   <none>           <none>
kube-system   pod/kube-controller-manager-ip-17               0/1     Running             1 (2m29s ago)   6m29s   2600:1f1c:ab4:ee48:c9fe:926d:609e:97e    ip-17    <none>           <none>
kube-system   pod/kube-controller-manager-ip-243              0/1     Running             0               117s    2600:iced:ab4:iced:coffee:baby:609e:97e   ip-243   <none>           <none>
kube-system   pod/kube-proxy-ip-229                           0/1     Running             0               22s     2600:iced:ab4:iced:coffee:baby:609e:97e   ip-229   <none>           <none>
kube-system   pod/kube-proxy-ip-17                            0/1     Running             1 (2m10s ago)   6m20s   2600:iced:ab4:iced:coffee:baby:609e:97e    ip-17    <none>           <none>
kube-system   pod/kube-proxy-ip-243                           0/1     Running             0               112s    2600:iced:ab4:iced:coffee:baby:609e:97e   ip-243   <none>           <none>
kube-system   pod/kube-proxy-ip-38                             0/1     Running             0               2m16s   2600:iced:ab4:iced:coffee:baby:609e:97e   ip-38     <none>           <none>
kube-system   pod/kube-scheduler-ip-229                       0/1     Running             0               25s     2600:iced:ab4:iced:coffee:baby:609e:97e   ip-229   <none>           <none>
kube-system   pod/kube-scheduler-ip-17                        0/1     Running             1 (2m29s ago)   6m29s   2600:iced:ab4:iced:coffee:baby:609e:97e    ip-17    <none>           <none>
kube-system   pod/kube-scheduler-ip-243                       0/1     Running             0               117s    2600:iced:ab4:iced:coffee:baby:609e:97e   ip-243   <none>           <none>
kube-system   pod/rke2-coredns-rke2-coredns-67f86d96c-2ljqt              1/1     Running             0               6m13s   2001:cafe:42::e0b                        ip-17    <none>           <none>
kube-system   pod/rke2-coredns-rke2-coredns-67f86d96c-9kqk4              1/1     Running             0               2m55s   2001:cafe:42:2::9139                     ip-243   <none>           <none>
kube-system   pod/rke2-coredns-rke2-coredns-autoscaler-d97d9cd9f-p44mv   1/1     Running             0               6m13s   2001:cafe:42::bdf3                       ip-17    <none>           <none>
kube-system   pod/rke2-ingress-nginx-controller-mbfxf                    1/1     Running             0               5m10s   2001:cafe:42::84a4                       ip-17    <none>           <none>
kube-system   pod/rke2-ingress-nginx-controller-sjkxf                    1/1     Running             0               89s     2001:cafe:42:2::2775                     ip-243   <none>           <none>
kube-system   pod/rke2-ingress-nginx-controller-xp8cv                    0/1     ContainerCreating   0               11s     <none>                                   ip-229   <none>           <none>
kube-system   pod/rke2-ingress-nginx-controller-z4nhx                    0/1     Running             0               55s     2001:cafe:42:3::8c77                     ip-38     <none>           <none>
kube-system   pod/rke2-metrics-server-c6fb46b64-sh6s2                    1/1     Running             0               5m23s   2001:cafe:42::1c01                       ip-17    <none>           <none>
kube-system   pod/rke2-snapshot-controller-59cc9cd8f4-pd87v              1/1     Running             0               5m19s   2001:cafe:42::f436                       ip-17    <none>           <none>
kube-system   pod/rke2-snapshot-validation-webhook-54c5989b65-fkdzt      1/1     Running             0               5m20s   2001:cafe:42::47f8                       ip-17    <none>           <none>

NAMESPACE     NAME                                              TYPE        CLUSTER-IP             EXTERNAL-IP   PORT(S)         AGE     SELECTOR
default       service/kubernetes                                ClusterIP   2001:cafe:42:1::1      <none>        443/TCP         6m40s   <none>
kube-system   service/cilium-agent                              ClusterIP   None                   <none>        9964/TCP        6m13s   k8s-app=cilium
kube-system   service/rke2-coredns-rke2-coredns                 ClusterIP   2001:cafe:42:1::a      <none>        53/UDP,53/TCP   6m13s   app.kubernetes.io/instance=rke2-coredns,app.kubernetes.io/name=rke2-coredns,k8s-app=kube-dns
kube-system   service/rke2-ingress-nginx-controller-admission   ClusterIP   2001:cafe:42:1::b8cc   <none>        443/TCP         5m10s   app.kubernetes.io/component=controller,app.kubernetes.io/instance=rke2-ingress-nginx,app.kubernetes.io/name=rke2-ingress-nginx
kube-system   service/rke2-metrics-server                       ClusterIP   2001:cafe:42:1::4ade   <none>        443/TCP         5m23s   app=rke2-metrics-server,release=rke2-metrics-server
kube-system   service/rke2-snapshot-validation-webhook          ClusterIP   2001:cafe:42:1::f371   <none>        443/TCP         5m20s   app.kubernetes.io/instance=rke2-snapshot-validation-webhook,app.kubernetes.io/name=rke2-snapshot-validation-webhook
```
